test/integration/http_server_options_test.lua

```diff
@@ -0,0 +1,75 @@
+local t = require('luatest')
+local http_client = require('http.client')
+
+local helpers = require('test.helpers')
+
+local g = t.group()
+
+g.before_test('test_keepalive_default', function()
+    g.httpd = helpers.cfgserv()
+    g.httpd:start()
+end)
+
+g.before_test('test_keepalive_change_in_runtime', function()
+    g.httpd = helpers.cfgserv()
+    g.httpd:start()
+end)
+
+g.before_test('test_keepalive_allowed', function()
+    g.httpd = helpers.cfgserv()
+    g.httpd:start()
+end)
+
+g.before_test('test_keepalive_disallowed', function()
+    local useragent = 'Mozilla/4.0'
+    g.httpd = helpers.cfgserv({
+        keepalive_disable = {
+            [useragent] = true,
+        }
+    })
+    g.httpd:start()
+end)
+
+g.after_each(function()
+    helpers.teardown(g.httpd)
+end)
+
+g.test_keepalive_default = function(g)
+    local httpd = g.httpd
+    t.assert_equals(httpd.options.keepalive_disable, {})
+end
+
+g.test_keepalive_allowed = function()
+    local useragent = 'Mozilla/4.0'
+    local r = http_client.request('GET', helpers.base_uri .. '/test', nil, {
+        headers = {
+            ['user-agent'] = useragent,
+            connection = 'keep-alive'
+        },
+    })
+    t.assert_equals(r.status, 200)
+    t.assert_equals(r.headers.connection, 'keep-alive')
+end
+
+g.test_keepalive_disallowed = function()
+    local useragent = 'Mozilla/4.0'
+    local r = http_client.request('GET', helpers.base_uri .. '/test', nil, {
+        headers = {
+            ['user-agent'] = useragent,
+            connection = 'keep-alive'
+        },
+    })
+    t.assert_equals(r.status, 200)
+    t.assert_equals(r.headers.connection, 'close')
+end
+
+g.test_keepalive_change_in_runtime = function(g)
+    local httpd = g.httpd
+    t.assert_equals(httpd.options.keepalive_disable, {})
+    httpd.options.keepalive_disable = {
+        ['Mozilla/4.0'] = true
+    }
+    t.assert_equals(httpd.options.keepalive_disable, {
+        ["Mozilla/4.0"] = true
+    })
```

Comment:
It is a bit strange. We assign into a table field and assert that it is there. Shouldn't we test that it is in effect?

Reply:
Now it is tested with internal handlers.